infra/templates/az-isolated-service-single-region/docs/design.md

```diff
@@ -0,0 +1,169 @@
+# Deploying App Service into ASE 
+
+v0.1 - 6/19/2019 
+
+## Introduction 
+
+Some of Cobalt’s enterprise customers have strict security guidelines that have led them to leverage Azure App Service Environments (ASE) as their primary infrastructure target when deploying to Azure. This is a natural choice as ASE enables customers to deploy to a single-tenant fleet of hosts running in a secure and isolated network that can be hooked into internal client-owned networks without traversing the public internet. 
+
+This document outlines how Cobalt can be extended to meet the use cases of these customers. The intended audience of this document is the development and product teams working on Cobalt and related projects. 
+
+## In Scope 
+
+- Identify deployment topology needed by the customer 
+- Identify key Terraform templates needed for deployment
+- Identify key Terraform modules needed for deployment
+- Identify gaps in Terraform provider templates
+
+## Out of scope 
+
+- External customer sign-off
+- Template (Terraform, ARM) implementation
+
+ 
+
+## Key Terms 
+- **RG**: Abbreviation for “Resource Group” 
+- **Sub**: Abbreviation for “Subscription” 
+- **Persona**: An archetype of a Cobalt customer 
+- **Stage**: An application deployment stage (dev, qa, pre-prod, prod, etc...) 
+- **Region**: A location in which an application is deployed 
+
+ 
+
+## Customers 
+- **Admin**: This persona represents an administrator of Azure. This persona does not implement the line of business applications but will help other teams deliver them. 
+- **App Developer Team**: This persona is responsible for creating and maintaining the line of business applications 
+
+## Open Questions 
+
+These open questions may impact the implementation plan and overall scope of the deployment described below 
+
+- What is the clients use case for Cosmos? Who manages these instances, and should we be automating the creation of them? Do we create the RG hosting Cosmos? If not, who owns this?
```

Comment:
Would be great to collect more clarity on what resource changes(if any) would the app dev team need to provision to azure?